phalcon/DataMapper/Query/AbstractConditions.zep

```diff
@@ -0,0 +1,358 @@
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalcon.io>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ *
+ * Implementation of this file has been influenced by AtlasPHP
+ *
+ * @link    https://github.com/atlasphp/Atlas.Query
+ * @license https://github.com/atlasphp/Atlas.Qyert/blob/1.x/LICENSE.md
+ */
+
+namespace Phalcon\DataMapper\Query;
+
+use Phalcon\Helper\Arr;
+
+/**
+ * Class AbstractConditions
+ */
+abstract class AbstractConditions extends AbstractQuery
+{
+    /**
+     * Sets the `LIMIT` clause
+     *
+     * @param int $limit
+     *
+     * @return AbstractConditions
+     */
+    public function limit(int limit) -> <AbstractConditions>
+    {
+        let this->store["LIMIT"] = limit;
+
+        return this;
+    }
+
+    /**
+     * Sets the `OFFSET` clause
+     *
+     * @param int $offset
+     *
+     * @return AbstractConditions
+     */
+    public function offset(int offset) -> <AbstractConditions>
+    {
+        let this->store["OFFSET"] = offset;
+
+        return this;
+    }
+
+    /**
+     * Sets a `AND` for a `WHERE` condition
+     *
+     * @param string     $condition
+     * @param mixed|null $value
+     * @param int        $type
+     *
+     * @return AbstractConditions
+     */
+    public function andWhere(
+        string condition,
+        var value = null,
+        int type = -1
+    ) -> <AbstractConditions> {
+        this->where(condition, value, type);
+
+        return this;
+    }
+
+    /**
+     * Concatenates to the most recent `WHERE` clause
+     *
+     * @param string     $condition
+     * @param mixed|null $value
+     * @param int        $type
+     *
+     * @return AbstractConditions
+     */
+    public function catWhere(
```

Comment:
Not sure if `catWhere` is the best naming for this. Maybe something like appendWhere?